All tests fail under `mvn package` with Maven 3.6.3

The `README` says Maven 3.9 is needed, but I'm stuck on 3.6.3, under which `mvn package` makes all tests fail, though they run fine in IntelliJ. JDK17 is being used in both cases.

```
wtwhite@wtwhite-vuw-vm:~/code$ git clone git@github.com:binaryeq/jmutator.git
Cloning into 'jmutator'...
--snip--
wtwhite@wtwhite-vuw-vm:~$ pushd ~/code/jmutator/
~/code/jmutator ~
wtwhite@wtwhite-vuw-vm:~/code/jmutator$ mvn package
[INFO] Scanning for projects...
[INFO] 
[INFO] --------------------< io.github.binaryeq:jmutator >---------------------
[INFO] Building jmutator 1.0.0
[INFO] --------------------------------[ jar ]---------------------------------
--snip--
-------------------------------------------------------
 T E S T S
-------------------------------------------------------
Running binaryeq.jmutator.ClassMutatorTest
Tests run: 9, Failures: 9, Errors: 0, Skipped: 0, Time elapsed: 0.025 sec <<< FAILURE!
binaryeq.jmutator.ClassMutatorTest.testMutationProvenanceFileNameTemplate2()  Time elapsed: 0.017 sec  <<< FAILURE!
org.opentest4j.AssertionFailedError: Unexpected exception type thrown, expected: <java.lang.IllegalArgumentException> but was: <java.lang.NullPointerException>
	at org.junit.jupiter.api.AssertionFailureBuilder.build(AssertionFailureBuilder.java:151)
	at org.junit.jupiter.api.AssertThrows.assertThrows(AssertThrows.java:67)
	at org.junit.jupiter.api.AssertThrows.assertThrows(AssertThrows.java:35)
	at org.junit.jupiter.api.Assertions.assertThrows(Assertions.java:3083)
	at binaryeq.jmutator.ClassMutatorTest.testMutationProvenanceFileNameTemplate2(ClassMutatorTest.java:127)
Caused by: java.lang.NullPointerException: Cannot invoke "java.io.File.exists()" because "binDir" is null
	at binaryeq.jmutator.ClassMutator.mutateClassFiles(ClassMutator.java:34)
	at binaryeq.jmutator.ClassMutatorTest.lambda$testMutationProvenanceFileNameTemplate2$9(ClassMutatorTest.java:127)
	at org.junit.jupiter.api.AssertThrows.assertThrows(AssertThrows.java:53)
	... 21 more

binaryeq.jmutator.ClassMutatorTest.testMutations()  Time elapsed: 0.001 sec  <<< FAILURE!
java.lang.NullPointerException: Cannot invoke "java.io.File.exists()" because "binDir" is null
	at binaryeq.jmutator.ClassMutator.mutateClassFiles(ClassMutator.java:34)
	at binaryeq.jmutator.ClassMutatorTest.testMutations(ClassMutatorTest.java:46)

binaryeq.jmutator.ClassMutatorTest.testValidityOfValidBytecode1()  Time elapsed: 0 sec  <<< FAILURE!
java.lang.NullPointerException: Cannot invoke "java.io.File.exists()" because "binDir" is null
	at binaryeq.jmutator.ClassMutator.mutateClassFiles(ClassMutator.java:34)
	at binaryeq.jmutator.ClassMutatorTest.testValidityOfValidBytecode1(ClassMutatorTest.java:59)

binaryeq.jmutator.ClassMutatorTest.testValidityOfValidBytecode2()  Time elapsed: 0.005 sec  <<< FAILURE!
java.lang.NullPointerException: Cannot invoke "java.io.File.exists()" because "binDir" is null
	at binaryeq.jmutator.ClassMutator.mutateClassFiles(ClassMutator.java:34)
	at binaryeq.jmutator.ClassMutatorTest.testValidityOfValidBytecode2(ClassMutatorTest.java:73)

binaryeq.jmutator.ClassMutatorTest.testMutationProvenance()  Time elapsed: 0 sec  <<< FAILURE!
java.lang.NullPointerException: Cannot invoke "java.io.File.exists()" because "binDir" is null
	at binaryeq.jmutator.ClassMutator.mutateClassFiles(ClassMutator.java:34)
	at binaryeq.jmutator.ClassMutatorTest.testMutationProvenance(ClassMutatorTest.java:88)

binaryeq.jmutator.ClassMutatorTest.testFindClassNames()  Time elapsed: 0 sec  <<< FAILURE!
java.lang.NullPointerException: Cannot invoke "java.io.File.toPath()" because "binDir" is null
	at binaryeq.jmutator.ClassMutator.findClassNames(ClassMutator.java:117)
	at binaryeq.jmutator.ClassMutatorTest.testFindClassNames(ClassMutatorTest.java:101)

binaryeq.jmutator.ClassMutatorTest.testMutatedClassNameTemplate1()  Time elapsed: 0 sec  <<< FAILURE!
org.opentest4j.AssertionFailedError: Unexpected exception type thrown, expected: <java.lang.IllegalArgumentException> but was: <java.lang.NullPointerException>
	at org.junit.jupiter.api.AssertionFailureBuilder.build(AssertionFailureBuilder.java:151)
	at org.junit.jupiter.api.AssertThrows.assertThrows(AssertThrows.java:67)
	at org.junit.jupiter.api.AssertThrows.assertThrows(AssertThrows.java:35)
	at org.junit.jupiter.api.Assertions.assertThrows(Assertions.java:3083)
	at binaryeq.jmutator.ClassMutatorTest.testMutatedClassNameTemplate1(ClassMutatorTest.java:109)
Caused by: java.lang.NullPointerException: Cannot invoke "java.io.File.exists()" because "binDir" is null
	at binaryeq.jmutator.ClassMutator.mutateClassFiles(ClassMutator.java:34)
	at binaryeq.jmutator.ClassMutatorTest.lambda$testMutatedClassNameTemplate1$6(ClassMutatorTest.java:109)
	at org.junit.jupiter.api.AssertThrows.assertThrows(AssertThrows.java:53)
	... 21 more

binaryeq.jmutator.ClassMutatorTest.testMutatedClassNameTemplate2()  Time elapsed: 0 sec  <<< FAILURE!
org.opentest4j.AssertionFailedError: Unexpected exception type thrown, expected: <java.lang.IllegalArgumentException> but was: <java.lang.NullPointerException>
	at org.junit.jupiter.api.AssertionFailureBuilder.build(AssertionFailureBuilder.java:151)
	at org.junit.jupiter.api.AssertThrows.assertThrows(AssertThrows.java:67)
	at org.junit.jupiter.api.AssertThrows.assertThrows(AssertThrows.java:35)
	at org.junit.jupiter.api.Assertions.assertThrows(Assertions.java:3083)
	at binaryeq.jmutator.ClassMutatorTest.testMutatedClassNameTemplate2(ClassMutatorTest.java:115)
Caused by: java.lang.NullPointerException: Cannot invoke "java.io.File.exists()" because "binDir" is null
	at binaryeq.jmutator.ClassMutator.mutateClassFiles(ClassMutator.java:34)
	at binaryeq.jmutator.ClassMutatorTest.lambda$testMutatedClassNameTemplate2$7(ClassMutatorTest.java:115)
	at org.junit.jupiter.api.AssertThrows.assertThrows(AssertThrows.java:53)
	... 21 more

binaryeq.jmutator.ClassMutatorTest.testMutationProvenanceFileNameTemplate1()  Time elapsed: 0 sec  <<< FAILURE!
org.opentest4j.AssertionFailedError: Unexpected exception type thrown, expected: <java.lang.IllegalArgumentException> but was: <java.lang.NullPointerException>
	at org.junit.jupiter.api.AssertionFailureBuilder.build(AssertionFailureBuilder.java:151)
	at org.junit.jupiter.api.AssertThrows.assertThrows(AssertThrows.java:67)
	at org.junit.jupiter.api.AssertThrows.assertThrows(AssertThrows.java:35)
	at org.junit.jupiter.api.Assertions.assertThrows(Assertions.java:3083)
	at binaryeq.jmutator.ClassMutatorTest.testMutationProvenanceFileNameTemplate1(ClassMutatorTest.java:121)
Caused by: java.lang.NullPointerException: Cannot invoke "java.io.File.exists()" because "binDir" is null
	at binaryeq.jmutator.ClassMutator.mutateClassFiles(ClassMutator.java:34)
	at binaryeq.jmutator.ClassMutatorTest.lambda$testMutationProvenanceFileNameTemplate1$8(ClassMutatorTest.java:121)
	at org.junit.jupiter.api.AssertThrows.assertThrows(AssertThrows.java:53)
	... 21 more


Results :

Failed tests:   binaryeq.jmutator.ClassMutatorTest.testMutationProvenanceFileNameTemplate2(): Unexpected exception type thrown, expected: <java.lang.IllegalArgumentException> but was: <java.lang.NullPointerException>
  binaryeq.jmutator.ClassMutatorTest.testMutations(): Cannot invoke "java.io.File.exists()" because "binDir" is null
  binaryeq.jmutator.ClassMutatorTest.testValidityOfValidBytecode1(): Cannot invoke "java.io.File.exists()" because "binDir" is null
  binaryeq.jmutator.ClassMutatorTest.testValidityOfValidBytecode2(): Cannot invoke "java.io.File.exists()" because "binDir" is null
  binaryeq.jmutator.ClassMutatorTest.testMutationProvenance(): Cannot invoke "java.io.File.exists()" because "binDir" is null
  binaryeq.jmutator.ClassMutatorTest.testFindClassNames(): Cannot invoke "java.io.File.toPath()" because "binDir" is null
  binaryeq.jmutator.ClassMutatorTest.testMutatedClassNameTemplate1(): Unexpected exception type thrown, expected: <java.lang.IllegalArgumentException> but was: <java.lang.NullPointerException>
  binaryeq.jmutator.ClassMutatorTest.testMutatedClassNameTemplate2(): Unexpected exception type thrown, expected: <java.lang.IllegalArgumentException> but was: <java.lang.NullPointerException>
  binaryeq.jmutator.ClassMutatorTest.testMutationProvenanceFileNameTemplate1(): Unexpected exception type thrown, expected: <java.lang.IllegalArgumentException> but was: <java.lang.NullPointerException>

Tests run: 9, Failures: 9, Errors: 0, Skipped: 0

[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  3.452 s
[INFO] Finished at: 2023-12-06T21:28:53+13:00
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-surefire-plugin:2.12.4:test (default-test) on project jmutator: There are test failures.
[ERROR] 
[ERROR] Please refer to /home/wtwhite/code/jmutator/target/surefire-reports for the individual test results.
[ERROR] -> [Help 1]
[ERROR] 
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR] 
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoFailureException
```

Versions:
```
wtwhite@wtwhite-vuw-vm:~/code/jmutator$ java --version
openjdk 17.0.8.1 2023-08-24
OpenJDK Runtime Environment (build 17.0.8.1+1-Ubuntu-0ubuntu122.04)
OpenJDK 64-Bit Server VM (build 17.0.8.1+1-Ubuntu-0ubuntu122.04, mixed mode, sharing)
wtwhite@wtwhite-vuw-vm:~/code/jmutator$ mvn --version
Apache Maven 3.6.3
Maven home: /usr/share/maven
Java version: 17.0.8.1, vendor: Private Build, runtime: /usr/lib/jvm/java-17-openjdk-amd64
Default locale: en_NZ, platform encoding: UTF-8
OS name: "linux", version: "5.15.0-87-generic", arch: "amd64", family: "unix"
```

--- a/pom.xml
+++ b/pom.xml
@@ -51,6 +51,11 @@
                     </execution>
                 </executions>
             </plugin>
+            <!-- Without this, Maven 3.6.3 defaults to version 2.12.4 which doesn't run @BeforeAll methods -->
+            <plugin>
+                <artifactId>maven-surefire-plugin</artifactId>
+                <version>2.22.1</version>
+            </plugin>
         </plugins>
     </build>
 

